src/customizations/before-send.ts

```diff
@@ -0,0 +1,69 @@
+import { clampToRange } from '../utils/number-utils'
+import { CaptureResult, KnownEventName } from '../types'
+import { includes } from '../utils'
+
+function simpleHash(str: string) {
+    let hash = 0
+    for (let i = 0; i < str.length; i++) {
+        hash = (hash << 5) - hash + str.charCodeAt(i) // (hash * 31) + char code
+        hash |= 0 // Convert to 32bit integer
+    }
+    return Math.abs(hash)
+}
+
+function sampleOnProperty(prop: string, percent: number): boolean {
+    return simpleHash(prop) % 100 < clampToRange(percent, 0, 100)
+}
+
+/**
+ * Provides an implementation of sampling that samples based on the distinct ID.
+ * Using the provided percentage.
+ * Can be used to create a beforeCapture fn for a PostHog instance.
+ *
+ * Causes roughly 50% of distinct ids to have events sent.
+ * Not 50% of events for each distinct id.
+ *
+ * @param percent a number from 0 to 100, 100 means never sample, 0 means never send the event
```

Comment:
yeah, i've always assumed it was laziness because math.random returns 0 to 1 rather than purposeful

but then this is laziness because the implementation underneath checks 0 to 100

I don't know what's best here... I guess if someone doesn't look at the signature then they'll send 0.1 expecting 10% and won't get that and we can't warn them because 0.1 is a valid number 🤔

Reply:
> I don't know what's best here... I guess if someone doesn't look at the signature then they'll send 0.1 expecting 10% and won't get that and we can't warn them because 0.1 is a valid number 🤔

that was exactly what I was afraid of but it's well-documented, not sure if auto-pilot reads comment blocks tho :D

Reply:
switched it to 0 to 1